### PR TITLE
use new ChildDefinition when possible

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.1
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolPass.php [hunk 1]
```diff
@@ -46,7 +46,7 @@ public function process(ContainerBuilder $container)
             if ($pool->isAbstract()) {
                 continue;
             }
-            while ($adapter instanceof DefinitionDecorator) {
+            while ($adapter instanceof DefinitionDecorator || $adapter instanceof ChildDefinition) {
```
Comment (ogizanagi) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolPass.php]: The use statement is missing

**Comment by xabbuh** [comment]
@ogizanagi missing `use` statement added

**Comment by ogizanagi** [comment]
👍 I saw that, hence the removal of my comment :) 
It's normal it fails on the `deps=high` build as it's not merged in upper branches yet, right?

Status: Reviewed

**Comment by xabbuh** [comment]
> I saw that, hence the removal of my comment :)

I was just wondering where it went. 😆 

> It's normal it fails on the deps=high build as it's not merged in upper branches yet, right?

Yes, for some PRs (like this one where a new class is added that is used in different components/bundles) the `deps=high` job is the only one that is allowed to fail.

**Comment by nicolas-grekas** [comment]
:-1:

**Pull request closed without merging**